src/main/java/org/ibp/api/rest/breedingmethod/BreedingMethodResource.java

```diff
@@ -0,0 +1,70 @@
+package org.ibp.api.rest.breedingmethod;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import org.generationcp.middleware.api.breedingmethod.BreedingMethodDTO;
+import org.generationcp.middleware.api.breedingmethod.BreedingMethodService;
+import org.generationcp.middleware.api.breedingmethod.MethodClassDTO;
+import org.generationcp.middleware.pojos.MethodGroup;
+import org.generationcp.middleware.pojos.MethodType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.Map;
+
+@Api(value = "Breeding methods Services")
+@RestController
+public class BreedingMethodResource {
+
+	@Autowired
+	private BreedingMethodService breedingMethodService;
+
+	@ApiOperation(value = "Get breeding method")
+	@RequestMapping(value = "/crops/{cropName}/breedingmethods/{breedingMethodDbId}", method = RequestMethod.GET)
+	@ResponseBody
+	public ResponseEntity<BreedingMethodDTO> getBreedingMethod(
+		@PathVariable final String cropName,
+		@PathVariable final Integer breedingMethodDbId,
+		@RequestParam(required = false) final String programUUID
+	) {
+		return new ResponseEntity<>(this.breedingMethodService.getBreedingMethod(breedingMethodDbId), HttpStatus.OK);
+	}
+
+	@ApiOperation(value = "List breeding method types")
+	@RequestMapping(value = "/crops/{cropName}/breedingmethod-types", method = RequestMethod.GET)
+	@ResponseBody
+	public ResponseEntity<List<MethodType>> getBreedingMethodTypes(
+		@PathVariable final String cropName,
+		@RequestParam(required = false) final String programUUID
+	) {
+		return new ResponseEntity<>(MethodType.getAll(), HttpStatus.OK);
+	}
```

Comment:
@clarysabel What do you think of that idea I told you once, about having XXXResourceGroup classes, to group resources that are closely related? So that we don't have to move them to different files? For this case I could rename the class to `BreedingMethodResourceGroup`

Reply:
I think it is fine but should not be better to name it XXXGroupResource? so we keep Resource at the end?